taxi/tests/test_views.py

```diff
@@ -0,0 +1,129 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase, Client
+from django.urls import reverse
+from taxi.models import Manufacturer, Car, Driver
+
+MANUFACTURER_LIST_URL = reverse("taxi:manufacturer-list")
+CAR_LIST_URL = reverse("taxi:car-list")
+CAR_DETAIL_URL = reverse("taxi:car-detail", kwargs={"pk": 1})
+DRIVER_LIST_URL = reverse("taxi:driver-list")
+DRIVER_DETAIL_URL = reverse("taxi:driver-detail", kwargs={"pk": 1})
+DRIVER_CREATE_URL = reverse("taxi:driver-create")
+
+
+class PublicManufacturerViewsTests(TestCase):
+
+    def test_login_required_manufacturer_list(self):
+        response = self.client.get(MANUFACTURER_LIST_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PublicCarViewsTests(TestCase):
+
+    def test_login_required_car_list(self):
+        response = self.client.get(CAR_LIST_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_car_detail(self):
+        response = self.client.get(CAR_DETAIL_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PublicDriverViewsTests(TestCase):
+
+    def test_login_required_driver_list(self):
+        response = self.client.get(DRIVER_LIST_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_driver_detail(self):
+        response = self.client.get(DRIVER_DETAIL_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivatManufacturerViewsTests(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.user = get_user_model().objects.create_user(
+            username="test", password="<PASSWORD>"
+        )
+        self.client.force_login(self.user)
+        Manufacturer.objects.create(
+            name="Test Manufacturer",
+            country="United Kingdom",
+        )
+        Manufacturer.objects.create(name="Test2 Manufacturer", country="USA")
+        self.response = self.client.get(MANUFACTURER_LIST_URL)
+
+    def test_login_required_manufacturer_list(self):
+        self.assertEqual(self.response.status_code, 200)
+
+    def test_retrieve_manufacturer_list(self):
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(self.response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(self.response, "taxi/manufacturer_list.html")
+
+
+class PrivatCarViewsTests(TestCase):
```

Comment:
The class name `PrivatCarViewsTests` should be `PrivateCarViewsTests` to maintain consistency and correct spelling.